connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java

```diff
@@ -214,6 +225,82 @@ public void testPoll() {
         }
     }
 
+    @Test
+    public void testSeekBehaviorDuringStart() {
+        // Setting up mock behavior.
+        @SuppressWarnings("unchecked")
+        KafkaConsumer<byte[], byte[]> mockConsumer = mock(KafkaConsumer.class);
+
+        @SuppressWarnings("unchecked")
+        KafkaProducer<byte[], byte[]> mockProducer = mock(KafkaProducer.class);
+
+        String sourceClusterName = "sourceCluster";
+        MirrorSourceMetrics mockMetrics = mock(MirrorSourceMetrics.class);
+
+        SourceTaskContext mockSourceTaskContext = mock(SourceTaskContext.class);
+        OffsetStorageReader mockOffsetStorageReader = mock(OffsetStorageReader.class);
+        when(mockSourceTaskContext.offsetStorageReader()).thenReturn(mockOffsetStorageReader);
+
+        MockedStatic<MirrorUtils> mockMirrorUtils = mockStatic(MirrorUtils.class, new CallsRealMethods());
+        mockMirrorUtils.when(() -> MirrorUtils.newConsumer(anyMap())).thenReturn(mockConsumer);
+        mockMirrorUtils.when(() -> MirrorUtils.newProducer(anyMap())).thenReturn(mockProducer);
+
+        Set<TopicPartition> topicPartitions = Sets.newSet(
```

Comment:
```suggestion
        Set<TopicPartition> topicPartitions = new HashSet<>(Arrays.asList(
```

(will also require a couple new imports and an additional trailing parenthesis)